database/migrations/000036_oci_prov.up.sql

```diff
@@ -0,0 +1,21 @@
+-- Copyright 2024 Stacklok, Inc
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- Add `ghcr` and `dockerhub` provider type enum
+ALTER TYPE provider_type ADD VALUE 'ghcr';
```

Comment:
Can you remind me of the difference between type and class?

Reply:
a type (or trait) is a claim of functionality. It denotes that you can do a certain subset of actions.

A class is like a template, it's a collection of types/traits alongside a collection of authentication flow. They allow us to aggregate these and build providers without having to always remember these magical combinations.